abstractions/php/src/Authentication/BaseBearerTokenAuthenticationProvider.php

```diff
@@ -6,19 +6,54 @@
 use Http\Promise\Promise;
 use Microsoft\Kiota\Abstractions\RequestInformation;
 
-abstract class BaseBearerTokenAuthenticationProvider implements AuthenticationProvider {
+/**
+ * Class BaseBearerTokenAuthenticationProvider
+ *
+ * Provides a base class for implementing {@link AuthenticationProvider} for Bearer authentication scheme
+ *
+ * @package Microsoft\Kiota\Abstractions\Authentication
+ * @copyright 2022 Microsoft Corporation
+ * @license https://opensource.org/licenses/MIT MIT License
+ * @link https://developer.microsoft.com/graph
+ */
+class BaseBearerTokenAuthenticationProvider implements AuthenticationProvider {
 
     /**
      * @var string $authorizationHeaderKey The Authorization header key
      */
     private static string $authorizationHeaderKey = "Authorization";
+
+    /**
+     * @var AccessTokenProvider {@link AccessTokenProvider}
+     */
+    private AccessTokenProvider $accessTokenProvider;
+
+    /**
+     * Creates a new instance
+     * @param AccessTokenProvider $accessTokenProvider to use for getting the access token
+     */
+    public function __construct(AccessTokenProvider $accessTokenProvider)
+    {
+        $this->accessTokenProvider = $accessTokenProvider;
```

Comment:
defensive programing, throw if null

Reply:
The argument type is not nullable. Null would be possible if the constructor argument type was `?AccessTokenProvider`
Welcome to PHP :)

Reply:
aaah cross cutting concerns implemented by the runtime/language! I love it, thanks for letting me know. resolving.